src/main/kotlin/com/openlattice/graph/partioning/RepartitioningJob.kt

```diff
@@ -0,0 +1,352 @@
+            setPartitions(state.entitySetId, state.newPartitions)
+            //The 4*getCount was an estimate, we remove estimate and addin updated value.
+            state.needsMigrationCount /= 2
+            state.needsMigrationCount += getNeedsMigrationCount()
+            state.currentlyMigratingPartitionIndex = 0
+            hasWorkRemaining = true
+            phase = 1
+        }
+    }
+
+    private fun delete(deleteSql: String): Long = hds.connection.use { connection ->
+        try {
+            connection.prepareStatement(deleteSql).use { deleteData ->
+                bind(deleteData)
+                deleteData.executeLargeUpdate()
+            }
+        } catch (ex: Exception) {
+            abort { "Job $id terminated: ${ex.message}" }
+            throw ex
+        } finally {
+            publishJobState()
+        }
+    }
+
+    override fun updateProgress() {
+        progress = ((100 * (state.repartitionCount + state.deleteCount)) / state.needsMigrationCount).toByte()
+    }
+
+    private fun repartition(repartitionSql: String): Long = hds.connection.use { connection ->
+        progress = ((100 * state.repartitionCount) / state.needsMigrationCount).toByte()
+        try {
+            connection.prepareStatement(repartitionSql).use { repartitionData ->
+                bind(repartitionData)
+                repartitionData.executeLargeUpdate()
+            }
+        } catch (ex: Exception) {
+            abort { "Job $id terminated: ${ex.message}" }
+            throw ex
+        } finally {
+            publishJobState()
+        }
+    }
+
+    private fun getCount(countSql: String, partition: Int): Long = hds.connection.use { connection ->
+        try {
+            connection.prepareStatement(countSql).use { countQuery ->
+                bind(countQuery, partition)
+                val rs = countQuery.executeQuery()
+                rs.next()
+                ResultSetAdapters.count(rs)
+            }
+        } catch (ex: Exception) {
+            abort { "Job $id terminated: ${ex.message}" }
+            throw ex
+        } finally {
+            publishJobState()
+        }
+    }
+
+    private fun getNeedsMigrationCount(): Long = 4 * state.oldPartitions.fold(0L) { count, partition ->
+        count + getCount(idsNeedingMigrationCountSql, partition) +
+                getCount(dataNeedingMigrationCountSql, partition) +
+                getCount(edgesNeedingMigrationCountSql, partition)
+    }
+
+
+    private fun bind(ps: PreparedStatement, partition: Int = currentlyMigratingPartition) {
+        ps.setObject(1, state.entitySetId)
+        ps.setArray(2, PostgresArrays.createIntArray(ps.connection, state.newPartitions))
+        ps.setInt(3, partition)
+    }
+
+    private fun setPartitions(entitySetId: UUID, partitions: Set<Int>) {
+        require(entitySets.containsKey(entitySetId)) {
+            "Entity set $entitySetId not found"
+        }
+        entitySets.executeOnKey(entitySetId) {
```

Comment:
I'm a little confused on this new syntax. Does this create an implicit serializer that can serialize `Set<Int>`? Or is there a requirement that only objects that are registered can be used here and hz just figures it out? 
Either way unless hz added `Set<Int>` as a built-in, partitions should probably be wrapped by a delegatedIntSet or whichever wrapper object we have for Set<Int>

Reply:
Kotlin lambda functions are java or closure serialized, so hazelcast won't see the captured Set<Int>. In this case, I figured that the rare invocation of the entry processor doesn't justify implementing a separate EntryProcessor class, StreamSerializer, and test. As a side note, if I was returning a Set<Int> I could wrap with DelegatedIntSet as the return values from the entry processor will be serialized using the traditional serialization mechanisms.

Reply:
Cool yeah the return wrapping makes sense. I don't know that I understand how serializing a lambda would avoid having to also serialize the `Set<Int>` to each node that executes this code though 🤔. If you're saying it *does* actually serialize the `Set<Int>` then cool, and I agree that seldom invoked java serializers are totally fine in this case.

Reply:
IIRC, lambdas are implemented as anonymous functions which include a closure of all captured variables that are out of scope. So there's basically a small class with a Set<Int> field that gets generated and is serialized using java/closure serialization (I'd have to look up if Kryo serialization is still configured). Most implementation of Set<Int> are serializable so this works. If partitions is invoked with non-serializable set then this will explode.

There's a couple of more rules to make sure that you don't accidentally capture the enclosing class for serialization, but that only happens when you invoke something on this or pass function reference to a class function.